### PR TITLE
docs: Update dependencies for documentation build system (Sphinx, add-ons etc.)

### PR DESCRIPTION
**Pull request by qmonnet**
The objectives of this PR are:

- Synchronising the rstcheck version between the minimal list of requirements and the list of actual dependencies for the docs build system,
- Making sure that the Google Analytics add-on is in that minimal list so that it doesn't disappear when regenerating the full list,
- Making sure reports from the new version of rstcheck are addressed,
- Updating (`make update-requirements`) the other dependencies, as much as we can.
- In particular, making sure we update markdown-it-py, for which [Dependabot reported there is a CVE](https://github.com/cilium/cilium/pull/23980) on our current version, even though I don't think we're affected.
- Remove the config for Dependabot, so it stops offering updates on Documentation/requirements-min/requirements.txt, to avoid getting out of sync again in the future.

Please refer to individual commits for details.

A new docs-builder image will be necessary, or the CI workflow will fail because of the old rstcheck not recognising the new syntax for its arguments.

### CODE REVIEW

**Comment by qmonnet** [comment]
Testing the changes: We need a new docs-builder image to properly test the changes. I built one, and pointed to it in a new (temporary) commit to make sure the CI workflow passes. It works (note the use of `qmonnet/docs-builder`):

image

The [Netlify previews](https://deploy-preview-24014--docs-cilium-io.netlify.app/) builds fine as well.

@joestringer, assuming the PR looks good to you, could you please pull the changes and push a new `docs-builder` image so I can update my last commit?

**Comment by qmonnet** [comment]
Added a commit to remove the Dependabot configuration for updates. We should no longer get PRs to updates Documentation/requirements-min/requirements.txt (we never merge them directly anyway), but should still get updates on Documentation/requirements.txt directly when Dependabot has a security advisory.

**Review by joestringer — COMMENTED** [review]
LGTM except for generating the new docs-builder image.

Seems like there's a similar issue with the docs-builder here as we have with the cilium-builder / cilium-runtime images. I agree for now it's OK to just handle this manually. If this bothers you, I would suggest looking a bit closer into (a) the way that the `images/` infrastructure detects changes and figures out canonical shas for a given set of input files, (b) the bot that can automatically build an image and update the commits in-line. Andre and I can help with any necessary docker repository configurations etc.

**Comment by joestringer** [comment]
If you push this as a PR directly to cilium/cilium, then it's a little easier to generate the image - effectively what I'm doing on the other side is just logging into dockerhub and then initiating a docs-builder image build from the dockerhub infrastructure. I think that previous times I have just copied this branch to the main repo in order to do this.

**Comment by joestringer** [comment]
(I can take care of this for now, so you don't have to reopen the PR)

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
       - name: Run pre-requisites for validation
         run: |
           make -C Documentation copy-api # necessary for check-build.sh
-      - uses: docker://cilium/docs-builder:2023-02-04@sha256:67c383332f63708fc40f3d12b0ffff8ec94a9a83d1b819dbe23b226ef076e6f2
+      - uses: docker://qmonnet/docs-builder:2023-02-24
```
Comment (joestringer) [inline .github/workflows/documentation.yaml]: cilium/docs-builder:2023-02-24 is on the way. I think you should be able to retrieve the sha from here once it's built? https://hub.docker.com/repository/registry-1.docker.io/cilium/docs-builder/builds/a4d7572f-ce1b-414b-9243-2e28d2f64815

**Review by joestringer — CHANGES_REQUESTED** [review]

**Comment by qmonnet** [comment]
> LGTM except for generating the new docs-builder image.

I agree the current process is not ideal. I didn't know it would make things simpler to push to the `cilium` repo, otherwise I'd have done so. But even that way, it's not optimal to ping you to generate a new image every time.

I don't intend to tackle this more in-depth in this PR, but I have opened https://github.com/cilium/cilium/issues/24019 to track this.

In the meantime, I've updated my commit to use the new image you've generated for this time -- Thanks a lot!

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by joestringer** [comment]
@qmonnet do you think that we need to backport this at all?

**Comment by qmonnet** [comment]
> @qmonnet do you think that we need to backport this at all?

I've been wondering, I would say no. This PR was mostly to get back to a “normal” state from where we can easily upgrade Sphinx and dependencies next time. I see no particular reason to backport and generate new docs-builder images for the older branches, for which the build is working well (and for which we don't expect to regenerate the list of build dependencies in the future), plus I doubt the backports would apply cleanly.

The only risk I see is if we try to backport other changes built on top of this PR in the future, like was done last time to 1.12, to some extent; but then we can still backport this PR if necessary when the time comes.

There is a [separate discussion regarding backports for Google Analytics](https://github.com/cilium/cilium/issues/22007#issuecomment-1445131132), though, but that doesn't need the rest of this PR in my opinion.

**Pull request merged**